[CI] AsyncSearchActionIT testRestartAfterCompletion failing

**Build Scans:**
- [elasticsearch-intake #40686 / part2](https://gradle-enterprise.elastic.co/s/gqqjnytpea22o)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:async-search:internalClusterTest" --tests "org.elasticsearch.xpack.search.AsyncSearchActionIT.testRestartAfterCompletion" -Dtests.seed=59771229DFFE3E05 -Dtests.locale=yrl -Dtests.timezone=SystemV/YST9YDT -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.search.AsyncSearchActionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRestartAfterCompletion'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: There are still [1] in-flight contexts. The first one's creation site is listed as the cause of this exception.
```

**Issue Reasons:**
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -573,12 +573,6 @@ tests:
   method: "builds distribution from branches via archives extractedAssemble [bwcDistVersion: 8.2.1, bwcProject: major3, expectedAssembleTaskName:
     extractedAssemble, #2]"
   issue: https://github.com/elastic/elasticsearch/issues/150410
-- class: org.elasticsearch.xpack.search.AsyncSearchActionIT
-  method: testRestartAfterCompletion
-  issue: https://github.com/elastic/elasticsearch/issues/150417
-- class: org.elasticsearch.xpack.search.AsyncSearchActionIT
-  method: testCCSCheckCompatibility
-  issue: https://github.com/elastic/elasticsearch/issues/150418
 - class: org.elasticsearch.xpack.dlm.frozen.DLMFrozenTransitionWithSecurityIT
   method: testCanTransitionIndex
   issue: https://github.com/elastic/elasticsearch/issues/149770

--- a/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchIntegTestCase.java
+++ b/x-pack/plugin/async-search/src/internalClusterTest/java/org/elasticsearch/xpack/search/AsyncSearchIntegTestCase.java
@@ -154,8 +154,13 @@ public abstract class AsyncSearchIntegTestCase extends ESIntegTestCase {
         ensureAllSearchContextsReleased();
 
         internalCluster().restartNode(node.getName(), new InternalTestCluster.RestartCallback() {});
-        unpauseMaintenanceService();
+        // Ensure shards are ready before unpausing the maintenance service. If we unpaused first,
+        // the maintenance service could dispatch a DeleteByQueryRequest to a shard that is still
+        // INITIALIZING, hitting the waitForSearchReady path. The resulting reader context would be
+        // registered after ensureAllSearchContextsReleased() passes, causing a spurious
+        // assertNoInFlightContext() failure in subsequent-test teardown.
         ensureYellow(ASYNC_RESULTS_INDEX, indexName);
+        unpauseMaintenanceService();
     }
 
     protected AsyncSearchResponse submitAsyncSearch(SubmitAsyncSearchRequest request) throws ExecutionException, InterruptedException {

